mesonbuild/interpreter.py

```diff
@@ -1753,7 +1756,7 @@ def override_find_program_method(self, args, kwargs):
             if not os.path.exists(abspath):
                 raise InterpreterException('Tried to override %s with a file that does not exist.' % name)
             exe = dependencies.ExternalProgram(abspath)
-        if not isinstance(exe, dependencies.ExternalProgram):
+        if not isinstance(exe, (dependencies.ExternalProgram, build.Executable)):
```

Comment:
Need to handle the case where `Executable` is actually cross-built. In that case we'll want to run it with the exe wrapper, or error out if an exe wrapper is not found.